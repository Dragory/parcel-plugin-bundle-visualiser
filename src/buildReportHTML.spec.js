const buildReportHTML = require('./buildReportHTML');

jest.mock('fs');


describe('./src/buildReportHTML', ()=>{

  it('should generate a report', ()=>{
    const mockTreeData = {
      groups:[
        { label:'thing1', weight:10 },
        { label:'thing2', weight:20 }
      ]
    };
    const report = buildReportHTML(mockTreeData);
    expect(report).toMatchSnapshot();
  });

});