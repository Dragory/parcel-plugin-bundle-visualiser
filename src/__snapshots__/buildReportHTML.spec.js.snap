// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./src/buildReportHTML should generate a report 1`] = `
"
  <!DOCTYPE html>
  <html>
    <head>
      <title>FoamTree Quick Start</title>
      <meta charset=\\"utf-8\\" />
    </head>
  
    <body>
      <div class=\\"container\\">
        <div id=\\"visualization\\"></div>
      </div>
      <style>
        body{ margin:0; }
        .container{ display:flex; height:100vh; }
        #visualization{ width:100%; }
      </style>
      <script>undefined</script>
      <script>

        var foamtree;

        window.addEventListener('load', function() {
          foamtree = new CarrotSearchFoamTree({
            id: 'visualization',
            dataObject: {\\"groups\\":[{\\"label\\":\\"thing1\\",\\"weight\\":10},{\\"label\\":\\"thing2\\",\\"weight\\":20}]},
            layoutByWeightOrder:true,
            layout: 'squarified',
            stacking: 'flattened',
            pixelRatio: window.devicePixelRatio || 1,
            maxGroupLevelsDrawn: Number.MAX_VALUE,
            maxGroupLevelsAttached:Number.MAX_VALUE,
            maxGroupLabelLevelsDrawn: Number.MAX_VALUE,
            rolloutDuration: 0,
            pullbackDuration: 0,
            fadeDuration: 0,
            zoomMouseWheelDuration: 300,
            openCloseDuration: 200,
            groupLabelVerticalPadding: 0.2,
            groupBorderRadius: 0,
            
            //TODO: follow up later
            // onGroupHover:function(e){
            //   console.log(evt.group && evt.group.label);
            //   if (e.group) {
            //     if (e.group.company) {
            //       detailsPanel.show(e.group.company);
            //     }
            //   } else {
            //     detailsPanel.hide();
            //   }
            // },

            //zoom to group rather than that weird pop out thing
            onGroupDoubleClick: function(e) {
              e.preventDefault();
              var group = e.group;
              var toZoom;
              if (group) {
                toZoom = e.secondary ? group.parent : group;
              } else {
                toZoom = this.get(\\"dataObject\\");
              }
              this.zoom(toZoom);
            }

          });
        });
  
        window.addEventListener(\\"resize\\", (function() {
          var timeout;
          return function() {
            window.clearTimeout(timeout);
            timeout = window.setTimeout(foamtree.resize, 300);
          };
        })());

      </script>
    </body>
  </html>
  "
`;
